Select language branch

tessmr opened the pull request:
Major Changes:
- GuessHandler.java
     - changed default return in matchYear from MY_GRAY to MY_YELLOW
     
- Added LanguageArray.java
     - Stores the list of languages and their data as objects; also stores the language objects in an array
     - Contains a method to get the array of languages
- Added SelectLanguage.java
     - Randomly selects a language from the array to choose as the correct guess
- Added ChoiceData.java
     - Stores the information of the correct guess and contains getters to retrieve the name, type, year, level, and paradigm (as an integer). 

Testing:
- Tested the change from gray to yellow, works as expected
- Had to test the three new files separately via print messages and they were able to randomly select a language and all the Getters in the ChoiceData returned the correct information. We need to call the select language method at the start of the game and have the ability to detect a win to fully test how these files integrate into the game.

tessmr committed:
Update GuessHandler.java

Changed the return color for matchYear to be MY_YELLOW if the year is incorrect to reflect the decision we made earlier today. Also updated the comment above the line of code to reflect that change.

tessmr committed:
Added three files to come up with a correct guess for the game. 

LanguageArray.java contains a list of objects for each programming language and its data. An array contains all the language objects and there is a method to retrieve that array.  

SelectLanguage.java randomly selects a programming language from the array provided by the LanguageArray class. 

ChoiceData.java stores the randomly selected object and contains getters to easily retrieve the information for the selected language.

RoboRobert commented on src/model/ChoiceData.java:
This file is useful, but is already mostly implemented in Guess.java. I would recommend just adding any extra data, like the integer for paradigm into that class. Probably we can just define a simple conversion function to convert between a string "PROCEDURAL" to 1, "STRUCTURED" to 2, etc.

You could also go ahead and refactor Guess.java to use private variables with getters. The data should really never change after initializing a Guess object, so it make sense here.

RoboRobert commented on src/model/SelectLanguage.java:
As long as this is reformatted to work with Guess objects instead of ChoiceData objects, it should work fine with what we already have implemented.

RoboRobert commented on src/model/LanguageArray.java:
This file looks good! Just a refactor to Guess objects instead of ChoiceData would make sense here. We also need to decide if we want to go the route of defining all our languages in code, or creating a JSON file to store them, which we could then read when necessary.
Also, making all the language variables into `private static final` makes sense here, since they should never change.
Either way, this will work great to test with.

RoboRobert reviewed (changes requested):
Overall, nice work!

tessmr committed:
Delete src/model/ChoiceData.java

Deleted ChoiceData because it is duplicative of Guess

tessmr committed:
Update Guess.java

Merged features (private variables, getters, and updated paradigm to an int) from ChoiceData into Guess

tessmr committed:
Update GuessHandler.java

Updated file to handle paradigm as an int instead of a string

tessmr committed:
Update LanguageArray.java

Changing each language variable to be private static final from just private.

tessmr committed:
Update GameFrame.java

Updated to use the getters in Guess rather than the direct variable and also to work with paradigm as an int.

tessmr commented:
Merged ChoiceData into Guess as requested. 
Removed ChoiceData
Updated LanguageArray to have all language variables private static final
Updated GuessHandler and GameFrame to use getters from Guess and work with paradigm as an integer. 

Will create a new branch to enumerate or create a conversion function to allow us to display the paradigm as its name. I will also handle paradigm comparison in that branch. 

When we are done testing we need to update the GameFrame to get correctAnswer from SelectLanguage.
